src/gui/torrentcontenttreeview.cpp

```diff
@@ -164,46 +163,7 @@ void TorrentContentTreeView::renameSelectedFile(BitTorrent::TorrentHandle *torre
         const QString oldPath {parentPath + oldName + '/'};
         const QString newPath {parentPath + newName + '/'};
 
-        // Check for overwriting
-#if defined(Q_OS_WIN)
-        const Qt::CaseSensitivity caseSensitivity = Qt::CaseInsensitive;
-#else
-        const Qt::CaseSensitivity caseSensitivity = Qt::CaseSensitive;
-#endif
-
-        for (int i = 0; i < torrent->filesCount(); ++i) {
-            const QString currentPath = torrent->filePath(i);
-
-            if (currentPath.startsWith(oldPath))
-                continue;
-
-            if (currentPath.startsWith(newPath, caseSensitivity)) {
-                RaisedMessageBox::warning(this, tr("The folder could not be renamed"),
-                                          tr("This name is already in use. Please use a different name."),
-                                          QMessageBox::Ok);
-                return;
-            }
-        }
-
-        // Replace path in all files
-        bool needForceRecheck = false;
-
-        for (int i = 0; i < torrent->filesCount(); ++i) {
-            const QString currentPath = torrent->filePath(i);
-
-            if (currentPath.startsWith(oldPath)) {
-                const QString path {newPath + currentPath.mid(oldPath.length())};
-
-                if (!needForceRecheck && QFile::exists(path))
-                    needForceRecheck = true;
-
-                torrent->renameFile(i, path);
-            }
-        }
-
-        // Force recheck
-        if (needForceRecheck)
-            torrent->forceRecheck();
+        torrent->renameFolder(oldPath, newPath);
```

Comment:
Seems you accidentally removed some logic from this method. It is unacceptable.

Reply:
Good catch. Initially the above logic was only present in one of the `TorrentHandleImpl::renameFolder` functions. Should it be put in both?

Reply:
I put the logic back.